<properties
 pageTitle="Connection timed out with Metastore client"
 description="Connection timed out with Metastore client"
 service="Microsoft.Databricks"
 resource="Microsoft.Databricks/workspaces"
 authors="AzureData_AzureDatabricks"
 ms.author="AzureData_AzureDatabricks"
 displayOrder=""
 selfHelpType="TSG_Content"
 supportTopicIds=""
 resourceTags=""
 productPesIds=""
 cloudEnvironments="public, fairfax, usnat, ussec"
 articleId="e958aa41-0f1f-4e39-8f4b-8cfb96f68b16"
 ownershipId="AzureData_AzureDatabricks"
/>

# Connection timed out with Metastore client

### Symptom

The Metastore client connection timed out with the following error:

`java.sql.SQLNonTransientConnectionException: Could not connect to address=(host=consolidated-centralus-prod-metastore.mysql.database.azure.com)(port=3306)(type=master) : Connection timed out (Connection timed out)
at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.get(ExceptionMapper.java:161)`

### Troubleshooting

- Do an **nc test** to verify that the metastore host is reachable from the customer VNet. Run similar command from a notebook in the customer environment:

		%sh nc -zv consolidated-centralus-prod-metastore.mysql.database.azure.com 3306

		%sh for i in {1..10} ; do nc -zv consolidated-westus-prod-metastore.mysql.database.azure.com 3306
		done

- Check if the customer has configured [UDR for allow listing metastore traffic and is using updated IP](https://docs.microsoft.com/en-us/azure/databricks/administration-guide/cloud-configurations/azure/udr).

- Check if you can reach metastore outside customer environment to confirm if it is customer environment specific issue.

- Try spinning up VM in customer Databricks subnet and see if it is reachable. Involve the Network team to troubleshoot on networking setup issues.


### Solution

- Resolve networking connectivity associated with the customer VNet setup. It can be due to firewall, SQL endpoint not allow listed, etc.

- If metastore traffic is allow listed using IP, make sure it uses the latest IP, as it can change over the time.

- If all the workspaces for specific region report metastore unavailable issue, it is MySQL DB issue for the region itself. File a Databricks Salesforce ticket.
